rgdps/repositories/user.py

```diff
@@ -47,6 +76,197 @@ async def create(ctx: Context, user: User) -> int:
+
+    user = User(
+        id=user_id,
+        username=username,
+        email=email,
+        password=password,
+        privileges=privileges,
+        message_privacy=message_privacy,
+        friend_privacy=friend_privacy,
+        comment_privacy=comment_privacy,
+        youtube_name=youtube_name,
+        twitter_name=twitter_name,
+        twitch_name=twitch_name,
+        register_ts=register_ts,
+        stars=stars,
+        demons=demons,
+        primary_colour=primary_colour,
+        secondary_colour=secondary_colour,
+        display_type=display_type,
+        icon=icon,
+        ship=ship,
+        ball=ball,
+        ufo=ufo,
+        wave=wave,
+        robot=robot,
+        spider=spider,
+        explosion=explosion,
+        glow=glow,
+        creator_points=creator_points,
+        coins=coins,
+        user_coins=user_coins,
+        diamonds=diamonds,
+    )
+
+    user.id = await ctx.mysql.execute(
+        "INSERT INTO users (id, username, email, password, privileges, message_privacy, "
+        "friend_privacy, comment_privacy, twitter_name, youtube_name, twitch_name, "
+        "stars, demons, primary_colour, secondary_colour, display_type, icon, "
+        "ship, ball, ufo, wave, robot, spider, explosion, glow, creator_points, "
+        "coins, user_coins, diamonds) VALUES (:id, :username, :email, :password, :privileges, "
+        ":message_privacy, :friend_privacy, :comment_privacy, :twitter_name, :youtube_name, "
+        ":twitch_name, :stars, :demons, :primary_colour, "
+        ":secondary_colour, :display_type, :icon, :ship, :ball, :ufo, :wave, :robot, "
+        ":spider, :explosion, :glow, :creator_points, :coins, :user_coins, :diamonds)",
+        user.as_dict(include_id=True),
+    )
+
+    return user
+
+
+async def update_partial(
+    ctx: Context,
+    user_id: int,
+    username: str | Unset = UNSET,
+    email: str | Unset = UNSET,
+    password: str | Unset = UNSET,
+    privileges: UserPrivileges | Unset = UNSET,
+    message_privacy: UserPrivacySetting | Unset = UNSET,
+    friend_privacy: UserPrivacySetting | Unset = UNSET,
+    comment_privacy: UserPrivacySetting | Unset = UNSET,
+    youtube_name: str | Unset = UNSET,
+    twitter_name: str | Unset = UNSET,
+    twitch_name: str | Unset = UNSET,
+    stars: int | Unset = UNSET,
+    demons: int | Unset = UNSET,
+    primary_colour: int | Unset = UNSET,
+    secondary_colour: int | Unset = UNSET,
+    display_type: int | Unset = UNSET,
+    icon: int | Unset = UNSET,
+    ship: int | Unset = UNSET,
+    ball: int | Unset = UNSET,
+    ufo: int | Unset = UNSET,
+    wave: int | Unset = UNSET,
+    robot: int | Unset = UNSET,
+    spider: int | Unset = UNSET,
+    explosion: int | Unset = UNSET,
+    glow: bool | Unset = UNSET,
+    creator_points: int | Unset = UNSET,
+    coins: int | Unset = UNSET,
+    user_coins: int | Unset = UNSET,
+    diamonds: int | Unset = UNSET,
+) -> User | None:
+    changed_data = {}
```

Comment:
nit: you'll eventually run into typing issues here if you don't explicitly type this at definition time in simpler cases - rn (if it's passing type checks), it is making a union of all the types you use later on
```suggestion
    changed_data: dict[str, Any] = {}
```

Reply:
oh actually it's just dict[Unknown, Unknown] if it's init'ed with no args - the breaking case is when you init it with something but don't specify the Any type for values,

![image](https://i.cmyui.xyz/bPmQNxcVRAl-vGrqcPdsQg.png)

Reply:
I have addressed this in places where this was an issue, should be fine here though?